### PR TITLE
Resolves the issues with +x on npm

### PR DESCRIPTION
**Pull request by EvanCarroll**
Works again with linux, resolves the issue with npm disabling -x by sourcing the local `xdg-open` through `/usr/bin/env`

### CODE REVIEW

**Comment by EvanCarroll** [comment]
This fixes #70

File: index.js [hunk 1]
```diff
@@ -47,14 +51,15 @@ module.exports = (target, opts) => {
...
+
 		if (opts.app) {
 			cmd = opts.app;
 		} else {
-			cmd = path.join(__dirname, 'xdg-open');
-		}
-
-		if (appArgs.length > 0) {
-			args = args.concat(appArgs);
+			cmd = path.join('/usr', 'bin', 'env');
```
Comment (kevva) [inline index.js]: I'd use something like [which](https://github.com/npm/node-which) here to ensure that we don't use the wrong path.

File: index.js [hunk 2]
```diff
@@ -47,14 +51,15 @@ module.exports = (target, opts) => {
...
+			args.unshift('xdg-open');
```
Comment (kevva) [inline index.js]: Why not `cmd = path.join('/usr', 'bin', 'env', 'xdg-open');`?

File: index.js [hunk 3]
```diff
@@ -47,14 +51,15 @@ module.exports = (target, opts) => {
 			args = args.concat(appArgs);
 		}
 	} else {
+		if (appArgs.length > 0) {
+			args = args.concat(appArgs);
+		}
```
Comment (kevva) [inline index.js]: Why did you move this?

**Comment by sindresorhus** [comment]
Thanks, but I rather just wait on npm to fix this. Looks like a fix there is close: https://github.com/zkat/pacote/pull/117

File: index.js (hunk 3 above)
Comment (EvanCarroll) [reply index.js]: Because unshifting (pushing to the front) of an array that isn't yet created is a bad idea.

File: index.js (hunk 2 above)
Comment (EvanCarroll) [reply index.js]: because xdg-open is the argument to `/usr/bin/env` there is no app called `/usr/bin/env/xdg-open`

**Comment by EvanCarroll** [comment]
That's no joy. Shipping out your own copy of `xdg-open` seems like a bad idea to begin with.
